dockstore-client/src/main/java/io/dockstore/common/FileProvisioning.java

```diff
@@ -355,6 +369,35 @@ private void provisionInputFile(String parameterFilePath, String targetPath, Pat
         }
     }
 
+    /**
+     * Returns the first target path and its scheme in <code>targetPaths</code>
+     * whose scheme is handled by a plugin in <code>pluginList</code>.
+     *
+     * @param pluginList     a list of plugins
+     * @param targetPaths    a list of targetPaths
+     * @return the first targetPath and scheme with a supported scheme, or <code>Optional.empty()</code>
+     */
+    static Optional<ImmutablePair<String, String>> findSupportedTargetPath(List<ProvisionInterface> pluginList, List<String> targetPaths) {
+        for (String target : targetPaths) {
+            try {
+                URI objectIdentifier = URI.create(target);    // throws IllegalArgumentException if it isn't a valid URI
```

Comment:
This seems correct, although it would be nice to have test cases that handle:

1. a target path where `URI.create()` throws an exception
2. A ProvisionInterface plugin that has two schemes handled, and that both schemes can match.